packages/injected/src/ariaSnapshot.ts

```diff
@@ -61,6 +69,9 @@ export function generateAriaTree(rootElement: Element, options?: { forAI?: boole
     visited.add(node);
 
     if (node.nodeType === Node.TEXT_NODE && node.nodeValue) {
+      if (node.parentElement && !isVisible(node.parentElement, options))
```

Comment:
Would be nice to solve it via a parameter boolean, that way you don't need to check parent visibility for every text child.

Reply:
Done. The behaviour around `ariaChildren` isn't super obvious to me, so I added a comment.